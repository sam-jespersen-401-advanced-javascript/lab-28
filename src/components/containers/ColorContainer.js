import React, { Component } from 'react';
import ColorDisplay from '../colors/ColorDisplay';
import ColorPicker from '../colors/ColorPicker';


export default class ColorContainer extends Component {
  state = {
    backgroundColor: 'black'
  };

  selectColor = backgroundColor => {
    this.setState({ backgroundColor });
  }

  render() {  
    const { backgroundColor } = this.state;
    const colors = ['red', 'blue', 'pink', 'purple', 'grey', 'orange', 'yellow', 'brown', 'green', 'white', 'black'];

    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} />
        <ColorPicker
          selectColor={this.selectColor}
          colors={colors} />
      </>
    );
  }

}
