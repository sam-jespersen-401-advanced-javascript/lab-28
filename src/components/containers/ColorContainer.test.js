import React from 'react';
import { shallow } from 'enzyme';
import ColorContainer from './ColorContainer';

describe('ColorContainer module', () => {
  it('renders a color container', () => {
    const wrapper = shallow(<ColorContainer />);
    expect(wrapper).toMatchSnapshot();
  });
});
