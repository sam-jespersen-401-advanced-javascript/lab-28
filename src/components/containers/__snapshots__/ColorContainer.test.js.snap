// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorContainer module renders a color container 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="black"
  />
  <ColorPicker
    colors={
      Array [
        "red",
        "blue",
        "pink",
        "purple",
        "grey",
        "orange",
        "yellow",
        "brown",
        "green",
        "white",
        "black",
      ]
    }
    selectColor={[Function]}
  />
</Fragment>
`;
